From 407036631b09ff916472b2e777eca433d603a6d2 Mon Sep 17 00:00:00 2001
From: Cristian Sovaiala <cristian.sovaiala@freescale.com>
Date: Thu, 18 Jun 2015 15:16:28 +0300
Subject: [PATCH 098/146] dpaa_eth: Use raw_smp_processor_id() to get queue
 mapping

Use raw_smp_processor_id() instead of smp_processor_id() to get
frame queue mapping. This is especially required on preemptible
kernels where preemption can change the CPU where the code is
executing on.

Signed-off-by: Cristian Sovaiala <cristian.sovaiala@freescale.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
index db52ede..40d956b 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
@@ -639,7 +639,7 @@ static inline void _dpa_assign_wq(struct dpa_fq *fq)
 
 #else
 #define dpa_get_queue_mapping(skb) \
-	smp_processor_id()
+	raw_smp_processor_id()
 #endif
 #else
 /* Use the queue selected by XPS */
-- 
2.1.0.27.g96db324

